src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,61 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function debounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), [
+    appliedQuery,
+  ]);
```

Comment:
The `useCallback` hook's dependency array should include `setAppliedQuery` to avoid potential stale closures. This ensures that the latest version of `setAppliedQuery` is used.